commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/CStandardTest.java

```diff
@@ -300,6 +300,107 @@ private static void assertConjugateEquality(Complex z,
+     * where the sign of the real/imaginary parts are unspecified in ISO C99. This is
+     * allowed (since they are unspecified). The sign specification is appropriately
+     * handled during testing of odd/even functions. There are some functions where it
+     * is not possible to satisfy the conjugate equality and also the odd/even rule.
+     * The compromise made here is to satisfy only one and the other is allowed to fail
+     * only on the sign of the output result. Known functions where this applies are:
+     *
+     * <ul>
+     *  <li>asinh(NaN, inf)
+     *  <li>atanh(NaN, inf)
+     *  <li>cosh(NaN, 0.0)
+     *  <li>sinh(inf, inf)
+     *  <li>sinh(inf, nan)
+     * </ul>
+     *
+     * @param operation the operation
+     */
+    private static void assertConjugateEquality(UnaryOperator<Complex> operation,
+                                                ComplexUnaryOperator<ComplexNumber> operation2) {
+        // Edge cases. Inf/NaN are specifically handled in the C99 test cases
+        // but are repeated here to enforce the conjugate equality even when the C99
+        // standard does not specify a sign. This may be revised in the future.
+        final double[] parts = {Double.NEGATIVE_INFINITY, -1, -0.0, 0.0, 1,
+            Double.POSITIVE_INFINITY, Double.NaN};
+        for (final double x : parts) {
+            for (final double y : parts) {
+                // No conjugate for imaginary NaN
+                if (!Double.isNaN(y)) {
+                    assertConjugateEquality(complex(x, y), operation, operation2, UnspecifiedSign.NONE);
+                }
+            }
+        }
+        // Random numbers
+        final UniformRandomProvider rng = RandomSource.SPLIT_MIX_64.create();
+        for (int i = 0; i < 100; i++) {
+            final double x = next(rng);
+            final double y = next(rng);
+            assertConjugateEquality(complex(x, y), operation, operation2, UnspecifiedSign.NONE);
+        }
+    }
+
+    /**
+     * Assert the operation on the complex number satisfies the conjugate equality.
+     *
+     * <pre>
+     * op(conj(z)) = conj(op(z))
+     * </pre>
+     *
+     * <p>The results must be binary equal; the sign of the complex number is first processed
+     * using the provided sign specification.
+     *
+     * Assert the operation on the complex number is <em>exactly</em> equal to the operation on
+     * complex real and imaginary parts.
+     *
+     * @param z the complex number
+     * @param operation1 the operation on the Complex object
+     * @param operation2 the operation on the complex real and imaginary parts
+     * @param sign the sign specification
+     */
+    private static void assertConjugateEquality(Complex z,
+        UnaryOperator<Complex> operation1,
+        ComplexUnaryOperator<ComplexNumber> operation2,
+        UnspecifiedSign sign) {
+
+        final Complex zConj = z.conj();
+        final Complex c1 = operation1.apply(zConj);
```

Comment:
Replace with:
```Java
final Complex c1 = TestUtils.assertSame(zConj, operation1, operation2, "");
final Complex c2 = TestUtils.assertSame(z, operation1, operation2, "").conj();
```

Then you do not need to generate ComplexNumber cn1 and cn2.

This possibly requires a name argument. It is missing from the original tests. I cannot recall why and so can you look into adding it so we know the conjugate equality that failed.